frontend/src/pages/NewRunV2.tsx

```diff
@@ -93,21 +93,33 @@ type NewRunV2Props = RunV2Props & PageProps;
 export type SpecParameters = { [key: string]: ComponentInputsSpec_ParameterSpec };
 export type RuntimeParameters = { [key: string]: any };
 
-function hasVersionID(
-  apiRun: ApiRunDetail | undefined,
-  apiRecurringRun: ApiJob | undefined,
-): boolean {
-  if (!apiRun && !apiRecurringRun) {
+type CloneOrigin = {
+  isClone: boolean;
+  isRecurring: boolean;
+  run: ApiRunDetail | undefined;
+  recurringRun: ApiJob | undefined;
+};
+
+function getCloneOrigin(apiRun: ApiRunDetail | undefined, apiRecurringRun: ApiJob | undefined) {
```

Comment:
NIT: We can use `function getCloneOrigin(apiRun?: ApiRunDetail, apiRecurringRun?: ApiJob)` to be concise.